### ISSUE TITLE
SingleInstanceService works correctly just for the first application run

### ISSUE THREAD

**Issue by vnashkev** [AdoptOpenJDK/IcedTea-Web#645]
After first start lock file with a port number is created 

C:\Users\user\AppData\Local\Temp\user\netx\locks\http___localhost_jnlp.jnlp.

Lock file still exists after application termination. 
Next time SingleInstanceServer fails to start with the following exception:
```
java.io.IOException: Cannot rename C:\Users\user\AppData\Local\Temp\user\netx\locks\http___localhost_jnlp.jnlp.temp to C:\Users\user\AppData\Local\Temp\user\netx\locks\http___localhost_jnlp.jnlp
	at net.adoptopenjdk.icedteaweb.io.FileUtils.createRestrictedFile(FileUtils.java:331)
	at net.adoptopenjdk.icedteaweb.io.FileUtils.createRestrictedFile(FileUtils.java:172)
	at net.sourceforge.jnlp.services.SingleInstanceLock.createWithPort(SingleInstanceLock.java:70)
	at net.sourceforge.jnlp.services.SingleInstanceLockTest.testCreateWithPort(SingleInstanceLockTest.java:34)
```

Test  case demonstraits the problem:
```
public final class SingleInstanceLockTest
	extends NoStdOutErrTest
{
    @Test
    public void testCreateWithPort() 
    	throws Exception 
    {
    	final URL url = this.getClass().getClassLoader().getResource("net/sourceforge/jnlp/basic.jnlp");
        final JNLPFile jnlpFile = new JNLPFile(url.openStream(), url, new ParserSettings(false,false,false));
        
        assert jnlpFile!=null;
        
        final SingleInstanceLock sil = new SingleInstanceLock(jnlpFile);
        
        sil.lockFile.delete();
        assert sil.lockFile.exists()==false;
        
        sil.createWithPort(123);
        assert sil.lockFile.exists()==true;
        assert "123".equals(FileUtils.loadFileAsUtf8String(sil.lockFile).trim());
        

        sil.createWithPort(456);
        assert sil.lockFile.exists()==true;
        assert "456".equals(FileUtils.loadFileAsUtf8String(sil.lockFile).trim());
    }
}
```

**Comment by janakmulani** [comment]
Thanks for reporting this issue.

I could reproduce your issue. We will fix it in a future release.

**Comment by vnashkev** [comment]
I have created pull request already :
[https://github.com/AdoptOpenJDK/IcedTea-Web/pull/646](url)

**Comment by sclassen** [comment]
fixed with #646 and #651
